packages/studio-base/src/components/ErrorDisplay.tsx

```diff
@@ -0,0 +1,144 @@
+    display: "flex",
+    flexDirection: "column",
+    flexGrow: 1,
+    overflow: "hidden",
+  },
+  errorDetailHeader: {
+    fontWeight: "bold",
+  },
+  errorDetailStack: {
+    fontSize: theme.typography.body2.fontSize,
+    lineHeight: "1.3em",
+    paddingLeft: theme.spacing(2),
+  },
+  errorDetailContainer: {
+    flexGrow: 2,
+    overflowY: "auto",
+  },
+  actions: {
+    paddingTop: theme.spacing(2),
+    textAlign: "right",
+  },
+}));
+
+function sanitizeStack(stack: string) {
+  return stack
+    .replace(/\s+\(.+\)$/gm, " (some location)")
+    .replace(/\s+https?:\/\/.+$/gm, " some location");
+}
+
+function ErrorStacktrace({ stack }: { stack: string }) {
+  const styles = useStyles();
+  const lines = sanitizeStack(stack)
+    .trim()
+    .replace(/^\s*at /gm, "")
+    .split("\n")
+    .map((line) => line.trim());
+  return (
+    <pre className={styles.errorDetailStack}>
+      {lines.map((line, i) => {
+        const match = /^(.+)\s(\(.+$)/.exec(line);
+        if (!match) {
+          return line + "\n";
+        }
+        return (
+          <span key={i}>
+            <span>{match[1]} </span>
+            <span>{match[2]}</span>
+            {"\n"}
+          </span>
+        );
+      })}
+    </pre>
+  );
+}
+
+type ErrorDisplayProps = {
+  title?: string;
+  error?: Error;
+  errorInfo?: ErrorInfo;
+  actions?: JSX.Element;
+  onDismiss?: () => void;
+};
+
+function ErrorDisplay(props: ErrorDisplayProps): JSX.Element {
+  const styles = useStyles();
+  const { error, errorInfo } = props;
+
+  const [showErrorDetails, setShowErrorDetails] = useState(false);
+
+  const errorDetails = useMemo(() => {
+    if (!showErrorDetails) {
+      return ReactNull;
+    }
+
+    let stackWithoutMessage = error?.stack ?? "";
+    const errorString = error?.toString() ?? "";
+    if (stackWithoutMessage.startsWith(errorString)) {
+      stackWithoutMessage = stackWithoutMessage.substring(errorString.length);
+    }
+
+    return (
+      <div>
+        <Typography className={styles.errorDetailHeader}>Error stack:</Typography>
+        <ErrorStacktrace stack={stackWithoutMessage} />
+        {errorInfo && (
+          <>
+            <Typography className={styles.errorDetailHeader}>Component stack:</Typography>
+            <ErrorStacktrace stack={errorInfo.componentStack} />
+          </>
+        )}
+      </div>
+    );
+  }, [error, errorInfo, showErrorDetails, styles]);
+
+  return (
+    <div className={styles.root}>
+      <div className={styles.alertContainer}>
+        <Typography variant="h4">
+          {props.title ?? "The app encountered an unexpected error"}
+        </Typography>
+        <AlertTitle></AlertTitle>
+        <Typography variant="body1">
+          <Stack spacing={2}>
+            <div>
+              <p>Something went wrong in the panel. Click Dismiss to continue using the panel.</p>
+              <p>If the issue persists try resetting the panel or removing the panel entirely.</p>
```

Comment:
```suggestion
              <p>An unexpected error occurred in this panel. Click Dismiss to continue using the panel. If the issue persists, try resetting the panel.</p>
```

Telling them to remove it doesn't seem that helpful, they can decide whether or not they want to keep the panel.

I find it a little weird to say "Click Dismiss to continue ..." directing them to a button when that text could've just been clickable itself. Maybe we make this text a link that also does the dismiss action?

Reply:
I'll make the text clickable that also does the dismiss action. I will remove the hint to remove the panel and keep the Remove panel button at the bottom.